gpu_memory_management/test/test.py

```diff
@@ -7,68 +7,90 @@
 
 #test 1 : there is two way to get the cupy pointer, but are they give a same result?
 def test_two_version_for_getting_cupy_pointer():
-    a = cp.ones(5)
+    cp.cuda.Device(0).use()
+    a = cp.array([3.14, 3.14, 3.14])
 
     assert(a.data.ptr == a.__cuda_array_interface__["data"][0])
 
 #test 2 : test if the size attribute get the right size
 def test_getting_cupy_size():
+    cp.cuda.Device(0).use()
     elem = 5   
-    a  = cp.ones(elem)
+    a  = cp.full(elem, 3.14)
 
     assert(elem == a.size)
 
 #test 3 : test if reinterpret cast with real cupy and custom cupy will result a same value
 def test_if_reinterpret_ptr_is_the_same():
-    a = cp.ones(5)
+    cp.cuda.Device(0).use()
+    a = cp.array([3.14, 3.14, 3.14])
     b = cupy_ref.Custom_Cupy_Ref(ptr = a.data.ptr, size = a.size)
     c = gpuMemManagement.test_if_reinterpret_ptr_is_the_same(a.data.ptr, b)
 
     assert(c == 1)
 
 #test 4 : test if the result of the  reinterpret cast of a real cupy pointer is a device pointer
 def test_if_real_cupy_reinterpret_ptr_is_a_gpu_array():
-    a = cp.ones(5)
+    cp.cuda.Device(0).use()
+    a = cp.array([3.14, 3.14, 3.14])
     res = gpuMemManagement.test_if_real_cupy_reinterpret_ptr_is_a_gpu_array(a.data.ptr)
 
     assert(res == 1)
 
 #test 5 : test if the pointer of a custom cupy pointer attributes is a device pointer
 def test_if_custom_cupy_reinterpret_ptr_is_a_gpu_array():
-    a = cp.ones(5)
+    cp.cuda.Device(0).use()
+    a = cp.array([3.14, 3.14, 3.14])
     b = cupy_ref.Custom_Cupy_Ref(ptr = a.data.ptr, size = a.size)
     res = gpuMemManagement.test_if_custom_cupy_reinterpret_ptr_is_a_gpu_array(b)
 
     assert(res == 1)
 
-#test 6 : copy array of ones from real cupy to cpu
-def test_copy_cupy_of_ones_to_cpu():
-    a = cp.ones(5)
-    b = gpuMemManagement.test_copy_cupy_of_ones_to_cpu(a.data.ptr, a.size)
+#test 6 : copy array of float from real cupy to cpu
+def test_copy_real_cupy_to_cpu():
+    cp.cuda.Device(0).use()
+    a = cp.array([3.14, 3.14, 3.14])
```

Comment:
I strongly recommend to not using a simple pattern for your test data, like repeating the same number or count from 0 until n. That data are corner cases and can provide different problems. For example, you want to implement a loop, which iterate over all elements of an array and do something with it. If you have the same value in all cells of your array, it is possible that you do a mistake in the loop and instead you loop over each element, you only loops over the first element N times (this happens more then one time for me ;-)). For the corner case of repeating numbers, the result is the same, but if you have different elements in the array, your algorithm will fail.

But this does not mean, that you need to replace all test data in all tests. If you only need a valid pointer or wants to check the size of an array, `np.ones()` is fine.